nativelib/src/main/resources/gc/immix/Stats.c

```diff
@@ -10,10 +10,18 @@ void Stats_Init(Stats *stats, const char *statsFile) {
     stats->collections = 0;
 }
 
-void Stats_RecordCollection(Stats *stats, uint64_t start_ns, uint64_t sweep_start_ns, uint64_t end_ns) {
+void Stats_RecordMark(Stats *stats, uint64_t start_ns,  uint64_t end_ns) {
     uint64_t index = stats->collections % STATS_MEASUREMENTS;
-    stats->mark_time_ns[index] = sweep_start_ns - start_ns;
-    stats->sweep_time_ns[index] = end_ns - sweep_start_ns;
+    stats->mark_time_ns[index] = end_ns - start_ns;
+    stats->sweep_time_ns[index] = 0L;
+}
+
+void Stats_RecordLazySweep(Stats *stats, uint64_t start_ns,  uint64_t end_ns) {
+    uint64_t index = stats->collections % STATS_MEASUREMENTS;
+    stats->sweep_time_ns[index] += end_ns - start_ns;
+}
+
+void Stats_RecordCollectionDone(Stats *stats) {
```

Comment:
I don't think aggregating here is appropriate. With real lazy sweeping we now have two types of pauses: one to mark and one to sweep a little. So pause time distribution should shift to the right with 'happy path' pause time being a cost of sweeping a few blocks. 

To get this info we need to log events like 'mark' and 'sweep' separately as they happen. So instead of:
```scala
mark_time,sweep_time
123,456
```
We should log events (m for mark and s for sweep):
```scala
m, 123
s, 5
s, 6
...
s, 4
```
Where all sweeps ad up to 456 for example.